app/src/main/java/org/wikipedia/settings/AppIconDialog.kt

```diff
@@ -0,0 +1,138 @@
+import com.google.android.flexbox.JustifyContent
+import org.wikipedia.LauncherController
+import org.wikipedia.LauncherIcon
+import org.wikipedia.R
+import org.wikipedia.appshortcuts.AppShortcuts
+import org.wikipedia.databinding.DialogAppIconBinding
+import org.wikipedia.databinding.ItemAppIconBinding
+import org.wikipedia.page.ExtendedBottomSheetDialogFragment
+import org.wikipedia.util.FeedbackUtil
+
+class AppIconDialog : ExtendedBottomSheetDialogFragment() {
+    private var _binding: DialogAppIconBinding? = null
+    private val binding get() = _binding!!
+
+    private val appIconAdapter: AppIconAdapter by lazy {
+        AppIconAdapter().apply {
+            onItemClickListener { selectedIcon ->
+                Prefs.currentSelectedAppIcon = selectedIcon.key
+                LauncherController.setIcon(selectedIcon)
+                AppShortcuts.setShortcuts(requireContext())
+                updateIcons(selectedIcon)
+                FeedbackUtil.makeSnackbar(
+                    binding.root,
+                    "App icon changed successfully.").show()
+            }
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = DialogAppIconBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupRecyclerView()
+    }
+
+    private fun setupRecyclerView() {
+        val layoutManager = FlexboxLayoutManager(requireContext()).apply {
+            flexDirection = FlexDirection.ROW
+            justifyContent = JustifyContent.CENTER
+            alignItems = AlignItems.CENTER
+        }
+        binding.appIconRecyclerView.apply {
+            this.layoutManager = layoutManager
+            adapter = appIconAdapter
+        }
+        appIconAdapter.updateItems(LauncherIcon.initialValues())
+    }
+
+    private fun updateIcons(selectedIcon: LauncherIcon) {
+        val currentSelectedIcon = if (Prefs.currentSelectedAppIcon != null) Prefs.currentSelectedAppIcon
+        else selectedIcon.key
+
+        LauncherIcon.entries.forEach {
+            it.isSelected = it.key == currentSelectedIcon
+        }
+        appIconAdapter.updateItems(LauncherIcon.entries)
+    }
+
+    private class AppIconAdapter : RecyclerView.Adapter<AppIconAdapter.AppIconViewHolder>() {
+        private var list = mutableListOf<LauncherIcon>()
+        private var onItemClickListener: ((LauncherIcon) -> Unit)? = null
+
+        fun onItemClickListener(onItemClickListener: (LauncherIcon) -> Unit) {
+            this.onItemClickListener = onItemClickListener
+        }
+
+        fun updateItems(newList: List<LauncherIcon>) {
+            list.clear()
+            list.addAll(newList)
+            notifyDataSetChanged()
```

Comment:
Very minor but we can apply `notifyItemRangeChanged(0, list.size)` here

Reply:
done.